test/sparse/sparse.jl

```diff
@@ -945,886 +1004,951 @@ function test_getindex_algs{Tv,Ti}(A::SparseMatrixCSC{Tv,Ti}, I::AbstractVector,
-
-# test similar with type conversion
-A = speye(5)
-@test size(similar(A,Complex128,Int)) == (5,5)
-@test typeof(similar(A,Complex128,Int)) == SparseMatrixCSC{Complex128,Int}
-@test size(similar(A,Complex128,Int8)) == (5,5)
-@test typeof(similar(A,Complex128,Int8)) == SparseMatrixCSC{Complex128,Int8}
-@test similar(A,Complex128,(6,6)) == spzeros(Complex128,6,6)
-@test convert(Matrix,A) == Array(A)
-
-# test float
-A = sprand(Bool, 5,5,0.0)
-@test eltype(float(A)) == Float64  # issue #11658
-A = sprand(Bool, 5,5,0.2)
-@test float(A) == float(Array(A))
-
-# test sparsevec
-A = sparse(ones(5,5))
-@test all(Array(sparsevec(A)) .== ones(25))
-@test all(Array(sparsevec([1:5;],1)) .== ones(5))
-@test_throws ArgumentError sparsevec([1:5;], [1:4;])
-
-#test sparse
-@test sparse(A) == A
-@test sparse([1:5;],[1:5;],1) == speye(5)
-
-#test speye and one
-@test speye(A) == speye(5)
-@test eye(A) == speye(5)
-@test one(A) == speye(5)
-@test_throws DimensionMismatch one(sprand(5,6,0.2))
-
-#istriu/istril
-A = sparse(triu(rand(5,5)))
-@test istriu(A)
-@test !istriu(sparse(ones(5,5)))
-A = sparse(tril(rand(5,5)))
-@test istril(A)
-@test !istril(sparse(ones(5,5)))
-
-# droptol
-srand(1234321)
-A = triu(sprand(10,10,0.2))
-@test Base.droptol!(A,0.01).colptr == [1,1,1,2,2,3,4,6,6,7,9]
-@test isequal(Base.droptol!(sparse([1], [1], [1]), 1), SparseMatrixCSC(1,1,Int[1,1],Int[],Int[]))
-
-# Test dropzeros[!]
-let smalldim = 5, largedim = 10, nzprob = 0.4, targetnumposzeros = 5, targetnumnegzeros = 5
-    for (m, n) in ((largedim, largedim), (smalldim, largedim), (largedim, smalldim))
-        A = sprand(m, n, nzprob)
-        struczerosA = find(x -> x == 0, A)
-        poszerosinds = unique(rand(struczerosA, targetnumposzeros))
-        negzerosinds = unique(rand(struczerosA, targetnumnegzeros))
-        Aposzeros = setindex!(copy(A), 2, poszerosinds)
-        Anegzeros = setindex!(copy(A), -2, negzerosinds)
-        Abothsigns = setindex!(copy(Aposzeros), -2, negzerosinds)
-        map!(x -> x == 2 ? 0.0 : x, Aposzeros.nzval)
-        map!(x -> x == -2 ? -0.0 : x, Anegzeros.nzval)
-        map!(x -> x == 2 ? 0.0 : x == -2 ? -0.0 : x, Abothsigns.nzval)
-        for Awithzeros in (Aposzeros, Anegzeros, Abothsigns)
-            # Basic functionality / dropzeros!
-            @test dropzeros!(copy(Awithzeros)) == A
-            @test dropzeros!(copy(Awithzeros), false) == A
-            # Basic functionality / dropzeros
-            @test dropzeros(Awithzeros) == A
-            @test dropzeros(Awithzeros, false) == A
-            # Check trimming works as expected
-            @test length(dropzeros!(copy(Awithzeros)).nzval) == length(A.nzval)
-            @test length(dropzeros!(copy(Awithzeros)).rowval) == length(A.rowval)
-            @test length(dropzeros!(copy(Awithzeros), false).nzval) == length(Awithzeros.nzval)
-            @test length(dropzeros!(copy(Awithzeros), false).rowval) == length(Awithzeros.rowval)
+@testset "diagm" begin
```

Comment:
Presumably there was a reason for the "on a matrix" in the comment, e.g. to contrast with tests elsewhere for `diagm` on scalars? Preserve the additional information in the comment?